src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -281,6 +281,19 @@ def __write_kv_and_features_to_file(file_path, key_values=None, features=None, f
         raise CLIError("Failed to export key-values to file. " + str(exception))
 
 
+# Exported in the format @Microsoft.AppConfiguration(Endpoint=<storeEndpoint>; Key=<kvKey>; Value=<kvValue>; Label=<kvLabel>).
+# Label is optional
+
+def __map_to_appservice_config_reference(key_value, endpoint):
+    if endpoint is None:
+        raise CLIError("No endpoint provided.")
+
+    label = key_value.label
+    key_value.value = AppServiceConstants.APPSVC_CONFIG_REFERENCE_PREFIX + 'Endpoint={0}; Key={1}'.format(
+        endpoint, key_value.key) + ('; Label={0}'.format(label) if label is not None else '') + ')'
```

Comment:
`key_value.key` has already been trimmed of the prefix that user might have specified with `--prefix` argument (we do this prefix trimming in `__read_kv_from_config_store`). 
While the AppSetting name is not impacted if the prefix is trimmed, this reference will not work since the trimmed key is not present in AppConfig.

I think this should be the behavior when --prefix and -r both are provided:

Lets say the prefix is "Beta_".

AppConfig key-value:
Key: Beta_Feature
Value: Test

After exporting to AppService:
Key: Feature
Value: @Microsoft.AppConfiguration(Endpoint=mystore.azconfig.io; Key=Beta_Feature)

Reply:
The simplest solution I can think of is pass the prefix to this method and append it to the value string appropriately.

Reply:
Since trimming will check whether the prefix exists before removing it, I move the trimming of the prefix to the `__map_to_appservice_config_reference` when the `export_as_reference` flag is indicated.